Source/Engine/Player.cs

```diff
@@ -1,23 +1,75 @@
-﻿using System.ComponentModel.DataAnnotations;
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+
 namespace EngineClasses
 {
     public class Player
     {
         [Key]
-        public int PlayerId { get; set; }
-        public string UserName { get; set; }
+        public int PlayerId { get; private set; }
+        public string UserName { get; private set; }
+        public string Color { get; private set; }
+
+        //Relationships
+        public int SessionId { get; private set; }
+        public Session Session { get; private set; }
+        public ICollection<GamePiece> GamePiece { get; private set; }
+
+        public Player(string userName, string color)
+        {
+            this.UserName = userName;
+            this.Color = color;
+            GamePiece = new List<GamePiece>();
+        }
 
-        public int SessionId { get; set; }
-        public Session Session { get; set; }
 
-        public void AddPlayerToDb(Player Player)
+        public int RollDice()
+        {
+            int result;
+            Random rnd = new Random();
+
+            result = rnd.Next(1, 6 + 1);
+            return result;
+        }
+
+        public GamePiece SelectGamePiece(int index)
+        {
+            return GamePiece.ToList()[index];
```

Comment:
To do this you will need to load all GamePieces, no problem when you only have 16....
I think the best solution would be to use .ElementAt(index) but it depends a bit what you are doing if index is the key in the database you should use .Find(index)